python/valis/routes/query.py

```diff
@@ -85,3 +85,19 @@ async def cone_search(self,
         """ Perform a cone search """
         return list(cone_search(ra, dec, radius, units=units))
 
+    @router.get('/carton_program_all', summary='Return a list of all carton or programs',
```

Comment:
In general, it's best practice to use simpler endpoint names.  Is there a common word that both carton and program could be called here and in the search endpoint that users would recognize?  We could use the description to clarify what the route does.  If not then we may want to convert this into two endpoints, one `/cartons`, and one `/programs`.   

Or maybe `/cartons-programs`, if we stick with the multi-word endpoint.  With multi-word endpoint names, it's also best practice to use hypens to improve readability. 

And maybe to differentiate from the other "search for target" endpoint, we should nest these under another level, maybe `/list/cartons-programs` or `/list/cartons` and `/list/programs`.  What do you think?

Reply:
I think this is a good idea! I created an implementation of this here: https://github.com/sdss/valis/pull/6/commits/9ae48924d61a162b8673e35eb08a62430868e7d4 Let me know what you think of it, as I wasn't 100% sure I was creating the new level correctly (e.g., unsure about how to construct the main query for the endpoint).

Reply:
Oh I was thinking of something much simpler.  Of keeping it under the main `query` route but adding a nested `list`.  So the full route would be something like `../valis/query/list/cartons`  or `../valis/query/list/programs`.  That a way a user could read this as "query a list of cartons".  API naming practices also suggest using plural nouns instead of singular, to indicate it returns a collection of resources.

Reply:
So something more like this: https://github.com/sdss/valis/pull/6/commits/d3ce33be3ecd4353c46de88656cdddca66f47ff1 ?

Reply:
yeah that looks good to me!